src/storage/txn/commands/compare_and_set.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::storage::kv::{Modify, WriteData};
+use crate::storage::lock_manager::LockManager;
+use crate::storage::raw;
+use crate::storage::txn::commands::{
+    Command, CommandExt, ResponsePolicy, TypedCommand, WriteCommand, WriteContext, WriteResult,
+};
+use crate::storage::txn::Result;
+use crate::storage::{ProcessResult, Snapshot};
+use engine_traits::CfName;
+use txn_types::{Key, Value};
+
+command! {
+    /// CompareAndSet to check whether the previous value of the key equals to the given value.
+    /// If they are equal, write the new value, otherwise return the previous value.
+    AtomicCompareAndSet:
```

Comment:
To match the name of protobuf:

```suggestion
    RawCompareAndSet:
```